### PR TITLE
fix: respect link case in link previews (WEBAPP-5174)

### PR DESCRIPTION
**Pull request by atomrc**

### CODE REVIEW

File: app/script/util/URLUtil.js [hunk 1]
```diff
@@ -75,7 +75,6 @@ z.util.URLUtil = (() => {
    */
   const _getDomainName = (url = '') => {
     return url
-      .toLowerCase()
```
Comment (atomrc) [inline app/script/util/URLUtil.js]: This function is only used to display the link preview url. So, unless I am missing something very important, I guess it's pretty safe to remove that lowercasing here...
@gregor @bennyn do you know the reason for lowercasing this?

Comment (bennycode) [reply app/script/util/URLUtil.js]: If tests are not failing, you are good to go! 🚦

Comment (gregor) [reply app/script/util/URLUtil.js]: The lowercasing was placed to catch cases like `WWW.WIRE.COM` and make sure we show `wire.com` instead. As tlds are case insensitive we should handle this properly and lowercase that part of the url.

File: app/script/util/URLUtil.js [hunk 2]
```diff
@@ -74,11 +74,14 @@ z.util.URLUtil = (() => {
    * @returns {string} Plain URL
    */
   const _getDomainName = (url = '') => {
-    return url
-      .toLowerCase()
-      .replace(/.*?:\/\//, '') // remove protocol
-      .replace(/\/$/, '') // remove trailing slash
-      .replace('www.', '');
+    const isMissingProtocol = url.length > 0 && !url.match(/^https?:\/\//i);
+    url = isMissingProtocol ? `http://${url}` : url;
```
Comment (AndyLnd) [inline app/script/util/URLUtil.js]: could also be written in a single line as
```
url = url.replace(/^(?!https?:\/\/)/i, 'http://');
```

**Pull request merged**